Composer/packages/tools/language-servers/language-generation/src/lgWorker.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Templates, Diagnostic } from 'botbuilder-lg';
+import { importResolverGenerator } from '@bfc/shared';
+
+import { WorkerMsg } from './lgParser';
+
+function createDiagnostic(diagnostic: Diagnostic) {
+  const { code, range, severity, source, message } = diagnostic;
+  const { start, end } = range;
+  return {
+    code,
+    range: {
+      start: { line: start.line, character: start.character },
+      end: { line: end.line, character: end.character },
+    },
+    severity,
+    source,
+    message,
+  };
+}
+
+// TODO: remove? not sure what this was for
+// if (!parentPort) {
+//   process.exit();
+// }
```

Comment:
This can probably be removed. I'm not really sure how it translates to the parent / child process model. This code should only be called once when the work is initialized so I don't think we need to check for an existing parent process.

Reply:
I removed the code